specs/src/specs/shares.md

```diff
@@ -0,0 +1,120 @@
+# Shares
+
+<!-- toc -->
+
+## Abstract
+
+All available data in a Celestia [block](./data_structures.md#block) is split into fixed-size data chunks known as "shares". Shares are the atomic unit of the Celestia data square. The shares in a Celestia block are eventually [erasure-coded](./data_structures.md#erasure-coding) and committed to in [Namespace Merkle trees](./data_structures.md#namespace-merkle-tree) (also see [NMT spec](https://github.com/celestiaorg/nmt/blob/master/docs/spec/nmt.md)).
+
+## Terms
+
+- **Blob**: User specified data (e.g. a roll-up block) that is associated with exactly one namespace. Blob data are opaque bytes of data that are included in the block but do not impact Celestia's state.
+- **Share**: A fixed-size data chunk that is associated with exactly one namespace.
+- **Share sequence**: A share sequence is a contiguous set of shares that contain semantically relevant data. A share sequence MUST contain one or more shares. When a [blob](../../../x/blob/README.md) is split into shares, it is written to one share sequence. As a result, all shares in a share sequence are typically parsed together because the original blob data may have been split across share boundaries. All transactions in the [`TRANSACTION_NAMESPACE`](./consensus.md#reserved-namespaces) are contained in one share sequence. All transactions in the [`PAY_FOR_BLOB_NAMESPACE`](./consensus.md#reserved-namespaces) are contained in one share sequence.
+
+## Overview
+
+User submitted [blob](../../../x/blob/README.md) data is split into shares (see [share splitting](#share-splitting)) and arranged in a `k * k` matrix (see [arranging available data into shares](./data_structures.md#arranging-available-data-into-shares)) prior to the erasure coding step. Shares in the `k * k` matrix are ordered by namespace and have a common [share format](#share-format).
```

Comment:
Ideally the share format is universal and applicable to all types of shares. The previous version of these specs differentiated between compact and sparse shares because that's how the current implementation behaves but IMO it should be easier to describe that all shares conform to one format (the "share format") and that transaction shares have additional metadata to support the use case of parsing a share in the middle of a sequence. WDYT?

Reply:
> that transaction shares have additional metadata to support the use case of parsing a share in the middle of a sequence. 

Thanks for the clarification. I see your point, that makes sense to me too. 
There are a few aspects that puzzled me:
- In the share format section, the supplied example is about the blob share, making this uncertainty that the explained share format only applies to the blob share (also given that there is no other section in the text dedicated to the blob share, it amplifies this confusion). One thing we could do (this PR or another PR) is to remove the blob share example from the "Share format" section and introduce a new section for "Blob Transactions", and there we explain how blob data is arranged in their own shares. 
- At the beginning of the "Transaction shares" section, the explanation somewhat implies the share format is specific to a namespace 
> In order for clients to parse shares in the middle of a sequence without downloading antecedent shares, **Celestia encodes additional metadata in the shares associated with reserved namespaces**. At the time of writing this only applies to the [TRANSACTION_NAMESPACE](https://github.com/celestiaorg/celestia-app/blob/6c7cc3ef6eb7d7f805556f7dccdfcf4fc7d90906/specs/src/specs/consensus.md#reserved-namespaces) and [PAY_FOR_BLOB_NAMESPACE](https://github.com/celestiaorg/celestia-app/blob/6c7cc3ef6eb7d7f805556f7dccdfcf4fc7d90906/specs/src/specs/consensus.md#reserved-namespaces).

However, that format applies to any transactions (but not blobs) where each individual transaction is a meaningful distinct unit of data to the Celestia blockchain, therefore, when a tx gets split across more than one share, additional metadata is needed to spot the start and end of that transaction. Maybe some rephrasing on that part can improve the clarity of that section.

Otherwise, looks good to me.